plugins/dicom_viewer/server/__init__.py

```diff
@@ -1,67 +1,129 @@
 from girder.api.rest import Resource
-from girder.constants import AccessType, TokenScope
-from girder.models.file import File
+from girder.constants import AccessType
 from girder.models.item import Item
+from girder.models.file import File
 
 
 class DicomItem(Resource):
 
-    @access.public(scope=TokenScope.DATA_READ)
+    @access.user
     @autoDescribeRoute(
-        Description('Get DICOM metadata, if any, for all files in the item.')
-        .modelParam('id', 'The item ID', model=Item, level=AccessType.READ, paramType='path')
-        .param('filters', 'Filter returned DICOM tags (comma-separated).',
-               required=False, default='')
-        .param('force', 'Force re-parsing the DICOM files. Write access required.',
-               required=False, dataType='boolean', default=False)
+        Description('Get and store common DICOM metadata, if any, for all files in the item.')
+        .modelParam('id', 'The item ID',
+                    model='item', level=AccessType.WRITE, paramType='path')
         .errorResponse('ID was invalid.')
         .errorResponse('Read permission denied on the item.', 403)
     )
-    def getDicom(self, item, filters, force):
-        if force:
-            Item().requireAccess(item, user=self.getCurrentUser(), level=AccessType.WRITE)
-        filters = set(filter(None, filters.split(',')))
-        files = list(Item().childFiles(item))
-        # process files if they haven't been processed yet
-        for i, f in enumerate(files):
-            if force or 'dicom' not in f:
-                files[i] = process_file(f)
-        # filter out non-dicom files
-        files = [x for x in files if x.get('dicom')]
-        # sort files
-        files = sorted(files, key=sort_key)
-        # execute filters
-        if filters:
-            for f in files:
-                dicom = f['dicom']
-                dicom = dict((k, dicom[k]) for k in filters if k in dicom)
-                f['dicom'] = dicom
-        return files
-
-
-def sort_key(f):
-    dicom = f.get('dicom', {})
+    def makeDicomItem(self, item):
+        """Try to convert an existing item into a "DICOM item", which contains a
+        "dicomMeta" field with DICOM metadata that is common to all DICOM files.
+        """
+        metadataReference = None
+        dicomFiles = []
+
+        for file in Item().childFiles(item):
+            dicomMeta = _parseFile(file)
+            if dicomMeta is None:
+                continue
+            dicomFiles.append(_extractFileData(file, dicomMeta))
+            dicomFiles.sort(key=_getDicomFileSortKey)
+            if metadataReference is None:
+                metadataReference = dicomMeta
+            else:
+                metadataReference = _removeUniqueMetadata(metadataReference, dicomMeta)
+
+        if len(dicomFiles) is not 0:
+            # Sort the dicom files
+            dicomFiles = sorted(dicomFiles, key=_getDicomFileSortKey)
```

Comment:
This will create a new list, rather than sort the existing one in place. Use `dicomFiles.sort` instead.